### PR TITLE
Add artifact manager for ballerina nodes

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
This is to support ballerina node integration to mi dashboard

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    public void runDeleteAllExecutorService() {
+        Runnable runnable = this::deleteAllArtifacts;
+        ExecutorServiceHolder.getBalArtifactsManagerExecutorService().execute(runnable);
+    }
+
+    private void fetchAndStoreServers(String accessToken) throws ManagementApiException {
+        // This code need to be changed according to the Ballerina server API
```
Comment (gabilang) [inline monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java:70]: We can remove this comment

File: monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        String url = heartbeat.getMgtApiUrl();
```
Comment (gabilang) [inline monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java:71]: With this URL we can only get nodeId and nodeData (bal-version, bal-home, os and os-version) right? We need to have another URL to get the artifact types (e.g., `String url = heartbeat.getMgtApiUrl() + "SERVICE";`)

Comment (HindujaB) [reply monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java:71]: yes. This is only for server information.

Comment (HindujaB) [reply monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java:71]: We might need to modify this further.

**Comment by HindujaB** [comment]
Closing as https://github.com/wso2/product-mi-tooling/pull/376 contains the updated changes.

**Pull request closed without merging**